### PR TITLE
Trace `CommunicationActions` into `Roles`

### PR DESCRIPTION
**Pull request by Prtfw**
## Description

Thank you for your contribution to the PySyft repository.
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

If your pull request closes a GitHub issue, then set its number below.

Fixes # (issue)

## Type of change

Please mark the options that are relevant.

- [ ] Added/Modified tutorials
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

* [x] I have added tests for my changes
* [x] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)
* [x] I have commented my code following [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).

### CODE REVIEW

File: syft/frameworks/torch/tensors/interpreters/native.py [hunk 1]
```diff
@@ -409,6 +410,7 @@ def send(
         preinitialize_grad: bool = False,
         no_wrap: bool = False,
         garbage_collect_data: bool = True,
+        torch: Any = None,
```
Comment (Jasopaum) [inline syft/frameworks/torch/tensors/interpreters/native.py:413]: What do you think we'll need that for?

File: syft/generic/pointers/pointer_tensor.py [hunk 2]
```diff
@@ -302,6 +302,7 @@ def remote_send(self, destination: AbstractWorker, requires_grad: bool = False):
 
     def remote_get(self):
         self.owner.send_command(message=("mid_get", self, (), {}), recipient=self.location)
+        self.child
```
Comment (karlhigley) [inline syft/generic/pointers/pointer_tensor.py:305]: Stray change?

File: syft/generic/pointers/pointer_tensor.py [hunk 3]
```diff
@@ -389,6 +388,9 @@ def share(self, *args, **kwargs):
         Returns:
             A pointer to an AdditiveSharingTensor
         """
+        if len(args) < 2:
+            raise RuntimeError("Error, share must have > 1 arguments all of type syft.workers")
```
Comment (karlhigley) [inline syft/generic/pointers/pointer_tensor.py:392]: 💯

**Comment by karlhigley** [comment]
🎉

File: syft/generic/pointers/pointer_tensor.py (hunk 2 above)
Comment (Prtfw) [reply syft/generic/pointers/pointer_tensor.py]: yep! sorry that one escaped me...

**Comment by Prtfw** [comment]
thanks y'all for the reviews and the merge ... 🙏

**Pull request merged**